File: rb/features.rb
# MemesioContentCreation SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module MemesioContentCreationFeatures
  def self.make_feature(name)
    case name
    when "base"
      MemesioContentCreationBaseFeature.new
    when "test"
      MemesioContentCreationTestFeature.new
    else
      MemesioContentCreationBaseFeature.new
    end
  end
end

File: py/memesiocontentcreation_sdk/features.py
# MemesioContentCreation SDK feature factory

from memesiocontentcreation_sdk.feature.base_feature import MemesioContentCreationBaseFeature
from memesiocontentcreation_sdk.feature.test_feature import MemesioContentCreationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MemesioContentCreationBaseFeature(),
        "test": lambda: MemesioContentCreationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
